third_party/terraform/resources/resource_logging_bucket_config.go

```diff
@@ -0,0 +1,143 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var loggingBucketConfigSchema = map[string]*schema.Schema{
+	"name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"location": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"bucket_id": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"description": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+	},
+	"retention_days": {
+		Type:     schema.TypeInt,
+		Optional: true,
+		Default:  30,
+	},
+	"lifecycle_state": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+type loggingBucketConfigIDFunc func(d *schema.ResourceData, config *Config) (string, error)
+
+// ResourceLoggingBucketConfig creates a resource definition by merging a unique field (eg: folder) to a generic logging bucket
+// config resource. In practice the only difference between these resources is the url location.
+func ResourceLoggingBucketConfig(parentSpecificSchema map[string]*schema.Schema, iDFunc loggingBucketConfigIDFunc) *schema.Resource {
+	// func ResourceLoggingBucketConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLoggingBucketConfigAcquire(iDFunc),
+		Read:   resourceLoggingBucketConfigRead(),
+		Update: resourceLoggingBucketConfigUpdate(),
+		Delete: resourceLoggingBucketConfigDelete(),
+		Schema: mergeSchemas(loggingBucketConfigSchema, parentSpecificSchema),
+	}
+}
+
+func resourceLoggingBucketConfigAcquire(iDFunc loggingBucketConfigIDFunc) func(*schema.ResourceData, interface{}) error {
+	return func(d *schema.ResourceData, meta interface{}) error {
+		config := meta.(*Config)
+
+		id, err := iDFunc(d, config)
+		if err != nil {
+			return err
+		}
+
+		d.SetId(id)
+
+		return resourceLoggingBucketConfigUpdate()(d, meta)
+	}
+}
+
+func resourceLoggingBucketConfigRead() schema.ReadFunc {
+	return func(d *schema.ResourceData, meta interface{}) error {
+		config := meta.(*Config)
+
+		log.Printf("[DEBUG] Fetching logging bucket config: %#v", d.Id())
+
+		url, err := replaceVars(d, config, fmt.Sprintf("{{LoggingBasePath}}%s", d.Id()))
+		if err != nil {
+			return err
+		}
+
+		res, err := sendRequest(config, "GET", "", url, nil)
+		if err != nil {
+			log.Printf("[WARN] Unable to acquire logging bucket config at %s", d.Id())
+
+			d.SetId("")
+			return err
```

Comment:
^

Reply:
strange, my comment here got lost. I was intentionally not using handleNotFound because it suppresses the error in addition to unsetting the ID. This means that a Create call with invalid params would appear to succeed but never set anything in state. Also since this type of resource is always present (non-destroyable) it should always be a hard error if there is a 404 since it means bad configuration.